src/components/activity-builder/edit-activity/step-builder/request-user-input-step-builder.tsx

```diff
@@ -433,6 +433,13 @@ export function RequestUserInputStepBuilder(props: {
             updateField('disableFreeInput', e);
           }}
         />
+        <CheckBoxInput
+          label="Set Student Activity Complete?"
+          value={step.setStudentActivityComplete ?? false}
+          onChange={(e) => {
+            updateField('setStudentActivityComplete', e);
```

Comment:
It looks like `setStudentActivityComplete` is not a function and it's passed as string to match something in an `if` or `switch`, but if so, you can use an `enum`

Reply:
Good catch. Although it can't be an enum. the `setStudentActivityComplete` is a field in an interface that I'm matching against. I update the updateField function to require that the field param is a key of the interface, like so:

`  function updateField<K extends keyof RequestUserInputActivityStep>(
    field: K,
    value: RequestUserInputActivityStep[K]
  ) `

This way, it confirms that he provided value is actually a ket of the interface, keeping it compile-time type-safe